source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -121,119 +121,134 @@ void GeoipProvider::lookup(Geolocation::LookupRequest&& request,
+                            MMDB_CITY_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->regionHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->regionHeader().value(),
+                            MMDB_REGION_LOOKUP_ARGS[0], MMDB_REGION_LOOKUP_ARGS[1],
+                            MMDB_REGION_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->countryHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->countryHeader().value(),
+                            MMDB_COUNTRY_LOOKUP_ARGS[0], MMDB_COUNTRY_LOOKUP_ARGS[1]);
+  }
+}
+
+void GeoipProvider::populateGeoLookupResultForIspDb(
+    MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->asnHeader().value(),
+                          MMDB_ASN_LOOKUP_ARGS[0]);
+}
+
+void GeoipProvider::populateGeoLookupResultForAnonDb(
+    MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  if (config_->isLookupEnabledForHeader(config_->anonHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[0]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonVpnHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonVpnHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[1]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonHostingHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonHostingHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonTorHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonTorHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[3]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonProxyHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonProxyHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[4]);
+  }
+}
+
+void GeoipProvider::populateGeoLookupResultForDbType(
+    const std::string& db_type, MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  if (db_type == "city") {
+    populateGeoLookupResultForCityDb(mmdb_lookup_result, lookup_result);
+  } else if (db_type == "isp") {
+    populateGeoLookupResultForIspDb(mmdb_lookup_result, lookup_result);
+  } else if (db_type == "anon") {
+    populateGeoLookupResultForAnonDb(mmdb_lookup_result, lookup_result);
+  }
+}
+
+void GeoipProvider::lookupInMaxmindDb(
+    const std::string& db_type, const std::string& stat_prefix,
```

Comment:
right now `stat_prefix` doesn't add much value since it's always the same as `db_type`, if we are ok with `db_type` using a string (vs. enum), then we could probably combine these two fields

Reply:
Don't think you need to pass/use `stat_prefix` here, as all maxmind stats is created in stat scope that already has stat_prefix as part of its name: https://github.com/envoyproxy/envoy/pull/31490/files#diff-d21c11080878cdf65727845da9fa3846d138efb810faeb12e8d16006713446deL30. Tagging @jmarantz as stats expert to confirm